File: java/src/test/java/fr/eulbobo/dojo/byron/domain/BriefRedoubtTest.java
package fr.eulbobo.dojo.byron.domain;

import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import static org.mockito.ArgumentMatchers.eq;
import static org.mockito.Mockito.*;

class BriefRedoubtTest {

    /**
     * | 1    | Time alone. Blissful time.                         | +1 Masterpiece                 |
     */
    @Test
    void shouldAdd1MasterPieceWhenRolls1(){
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new BriefRedoubt(eventLog);
        event.applyDiceTo(DiceRoll.of(1), sanity);

        verify(eventLog).log(eq("Time alone. Blissful time."));
        verify(sanity, times(1)).increaseMasterpiece();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * | 2    | He's busy with a paramour.                         | -1 Stress                      |
     */
    @Test
    void shouldDecrease1StressWhenRolls2(){
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new BriefRedoubt(eventLog);
        event.applyDiceTo(DiceRoll.of(2), sanity);

        verify(eventLog).log(eq("He's busy with a paramour."));
        verify(sanity, times(1)).decreaseStress();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * | 3    | A walk around the house ! Underwear on our heads ! | +1 Scandal                     |
     */
    @Test
    void shouldIncrease1ScandalWhenRolls3(){
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new BriefRedoubt(eventLog);
        event.applyDiceTo(DiceRoll.of(3), sanity);

        verify(eventLog).log(eq("A walk around the house ! Underwear on our heads !"));
        verify(sanity, times(1)).increaseScandal();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * | 4    | He has an excellent supply of contraband substance | +1 Scandal <br> +1 Masterpiece |
     */
    @Test
    void shouldIncrease1ScandalAndMasterpieceWhenRolls4(){
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new BriefRedoubt(eventLog);
        event.applyDiceTo(DiceRoll.of(4), sanity);

        verify(eventLog).log(eq("He has an excellent supply of contraband substance"));
        verify(sanity, times(1)).increaseScandal();
        verify(sanity, times(1)).increaseMasterpiece();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * | 5    | Wine ! A chest of wine !                           | -1 Masterpiece                 |
     */
    @Test
    void shouldDecreaseMasterpieceWhenRolls5(){
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new BriefRedoubt(eventLog);
        event.applyDiceTo(DiceRoll.of(5), sanity);

        verify(eventLog).log(eq("Wine ! A chest of wine !"));
        verify(sanity, times(1)).decreaseMasterpiece();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * | 6    | He passed out in his study                         | +1 Masterpiece                 |
     */
    @Test
    void shouldIncreaseMasterpieceWhenRolls6(){
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new BriefRedoubt(eventLog);
        event.applyDiceTo(DiceRoll.of(6), sanity);

        verify(eventLog).log(eq("He passed out in his study"));
        verify(sanity, times(1)).increaseMasterpiece();
        verifyNoMoreInteractions(sanity);
    }
}

File: Csharp/fr.eulbobo.dojo.byron/fr.eulbobo.dojo.byron.tests/BriefRedoubtTest.cs
using fr.eulbobo.dojo.byron.domain;
using Moq;

namespace fr.eulbobo.dojo.byron.tests
{
    public class BriefRedoubtTest
    {

        /**
         * | 1    | Time alone. Blissful time.                         | +1 Masterpiece                 |
         */
        [Test]
        public void ShouldAdd1MasterPieceWhenRolls1()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event = new BriefRedoubt(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(1), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("Time alone. Blissful time."));
            sanity.Verify(s => s.IncreaseMasterpiece(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * | 2    | He's busy with a paramour.                         | -1 Stress                      |
         */
        [Test]
        public void ShouldDecrease1StressWhenRolls2()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event = new BriefRedoubt(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(2), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("He's busy with a paramour."));
            sanity.Verify(s => s.DecreaseStress(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * | 3    | A walk around the house ! Underwear on our heads ! | +1 Scandal                     |
         */
        [Test]
        public void ShouldIncrease1ScandalWhenRolls3()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event = new BriefRedoubt(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(3), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("A walk around the house ! Underwear on our heads !"));
            sanity.Verify(s => s.IncreaseScandal(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * | 4    | He has an excellent supply of contraband substance | +1 Scandal <br> +1 Masterpiece |
         */
        [Test]
        public void ShouldIncrease1ScandalAndMasterpieceWhenRolls4()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event = new BriefRedoubt(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(4), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("He has an excellent supply of contraband substance"));
            sanity.Verify(s => s.IncreaseScandal(), Times.Once);
            sanity.Verify(s => s.IncreaseMasterpiece(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * | 5    | Wine ! A chest of wine !                           | -1 Masterpiece                 |
         */
        [Test]
        public void ShouldDecreaseMasterpieceWhenRolls5()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event = new BriefRedoubt(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(5), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("Wine ! A chest of wine !"));
            sanity.Verify(s => s.IncreaseMasterpiece(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * | 6    | He passed out in his study                         | +1 Masterpiece                 |
         */
        [Test]
        public void ShouldIncreaseMasterpieceWhenRolls6()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event = new BriefRedoubt(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(6), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("He passed out in his study"));
            sanity.Verify(s => s.IncreaseMasterpiece(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }
    }
}